Lib/multiprocessing/connection.py

```diff
@@ -734,30 +735,93 @@ def PipeClient(address):
 WELCOME = b'#WELCOME#'
 FAILURE = b'#FAILURE#'
 
-def deliver_challenge(connection, authkey):
+_mac_algo_re = re.compile(
+    rb'^{(?P<digestmod>(md5|sha256|sha384|sha3_256|sha3_384))}'
+    rb'(?P<payload>.*)$'
```

Comment:
How long can the payload be? Perhaps it's overkill (and potentially costly) to make this a specific capture group instead of simply taking the rest of the string after `digestmod`.